File: services/assessment-service/src/main/java/com/visor/school/assessment/repository/ReportSubmissionRepository.java
package com.visor.school.assessment.repository;

import com.visor.school.assessment.model.ReportSubmission;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface ReportSubmissionRepository extends JpaRepository<ReportSubmission, UUID> {
    List<ReportSubmission> findByClassId(UUID classId);
    List<ReportSubmission> findBySubmittedBy(UUID submittedBy);
    List<ReportSubmission> findByCollectionId(UUID collectionId);
}

File: services/academic-assessment-service/src/main/kotlin/com/visor/school/assessment/repository/ReportSubmissionRepository.kt
package com.visor.school.assessment.repository

import com.visor.school.assessment.model.ReportSubmission
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface ReportSubmissionRepository : JpaRepository<ReportSubmission, UUID> {
    fun findByClassId(classId: UUID): List<ReportSubmission>
    fun findBySubmittedBy(submittedBy: UUID): List<ReportSubmission>
    fun findByCollectionId(collectionId: UUID): List<ReportSubmission>
}

